src/main/java/com/purbon/kafka/topology/api/connect/KConnectApiClient.java

```diff
@@ -46,14 +49,26 @@ public Map<String, Object> add(String content) throws IOException {
   }
 
   public List<String> list() throws IOException {
-    Response response = doGet("/connectors");
-    return JSON.toArray(response.getResponseAsString());
+    JsonNode node = doList();
+    return ImmutableList.copyOf(node.fieldNames());
+  }
+
+  protected JsonNode doList() throws IOException {
+    Response response = doGet("/connectors?expand=info");
+    return JSON.toNode(response.getResponseAsString());
   }
 
   @Override
   public Collection<? extends Artefact> getClusterState() throws IOException {
-    return list().stream()
-        .map(connector -> new KafkaConnectArtefact("", server, connector))
+    JsonNode list = doList();
+    Iterable<Map.Entry<String, JsonNode>> fields = list::fields;
+    return StreamSupport.stream(fields.spliterator(), false)
+        .map(
+            entry -> {
+              JsonNode config = entry.getValue().get("config");
```

Comment:
Be carefull, you have to replace by : 

``` java
JsonNode config = entry.getValue().get("info").get("config");
```

Here an example payload from the connect cluster at http://localhost:8083/connectors?expand=info : 

``` json
{
    "source-datagen": {
        "info": {
            "name": "source-datagen",
            "config": {
                "connector.class": "io.confluent.kafka.connect.datagen.DatagenConnector",
                "quickstart": "stock_trades",
                "tasks.max": "1",
                "name": "source-datagen",
                "kafka.topic": "context.projectA.source"
            },
            "tasks": [
                {
                    "connector": "source-datagen",
                    "task": 0
                }
            ],
            "type": "source"
        }
    }
}
```

Reply:
thanks, fixed that